### DISCUSSION TITLE
public api add a relationship

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#5678] [General]
Hello,

I hope you are doing well.

I am currently trying your api and using your readme app. A simple question how can I add a relationship in my rows?

thank you

**Comment by melohagan** [comment]
Hello @oswaldB 

Here's the docs on relationships: https://docs.budibase.com/docs/relationships

If you want to display the columns of a relationship in the same table, then you can use formulas: https://docs.budibase.com/docs/formula#fields-from-relationships

**Comment by melohagan** [comment]
Sorry @oswaldB 

I initially misread your question, you want to handle relationships with the API. 

If you do a GET on a table with a relationship column, you'll see something like this:
![Screenshot 2022-05-03 at 08 42 20](https://user-images.githubusercontent.com/101575380/166419152-6aae2838-899b-41ca-af65-9014e92d472c.png)

You can use the same payload in your PUT to update an existing table to add a *link* type column if it does not already exist.

To update or add rows with a relationship, you can do a PUT or a POST with the name of your relationship column like so:
![Screenshot 2022-05-03 at 08 47 15](https://user-images.githubusercontent.com/101575380/166419660-aa7ef9d2-56c9-49d5-8d4a-6b56c5d1018b.png)
In this case you'd replace *relationship* with the field name: product

https://docs.budibase.com/reference/post_tables-tableid-rows
